packages/views/src/errorBoundary.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+
+export type FallbackRender = (errorData: {
+  error: Error;
+  errorInfo: React.ErrorInfo | null;
+  resetError(): void;
+}) => React.ReactElement;
+
+export interface ErrorBoundaryProps {
+  onError?: (error: Error, errorInfo: React.ErrorInfo) => void;
+  /** Called if resetError() is called from the fallback render props function  */
+  onReset?(error: Error | null, errorInfo: React.ErrorInfo | null): void;
+  fallback?: React.ReactElement | FallbackRender;
+}
+
+interface ErrorBoundaryState {
+  error: Error | null;
+  errorInfo: React.ErrorInfo | null;
+}
+
+const INITIAL_STATE = {
+  errorInfo: null,
+  error: null,
+};
+
+export default class ErrorBoundary extends React.PureComponent<ErrorBoundaryProps, ErrorBoundaryState> {
+  public state: ErrorBoundaryState = INITIAL_STATE;
+
+  public resetErrorBoundary: () => void = () => {
+    const { onReset } = this.props;
+    const { error, errorInfo } = this.state;
```

Comment:
Can we init the variables only when they are needed (inside the following `if` statement)?